src/Utils/BlockTemplateUtils.php

```diff
@@ -133,24 +133,61 @@ public static function gutenberg_build_template_result_from_post( $post ) {
 	 */
 	public static function gutenberg_build_template_result_from_file( $template_file, $template_type ) {
 		$template_file = (object) $template_file;
+
+		// If the theme has an archive-products.html template but does not have product taxonomy templates
+		// then we will load in the archive-product.html template from the theme to use for product taxonomies on the frontend.
+		$template_is_from_theme = 'theme' === $template_file->source ? true : false;
```

Comment:
nit: `'theme' === $template_file->source` evaluates to boolean, so the `true : false` is not needed here.